### ISSUE TITLE
Politeness suffix сĂм

### ISSUE THREAD

**Issue by hectoralos** [apertium/apertium-chv#17]
I've tried to introduce this suffix, but I've done it wrong and I don't know how I should do it.

There are 3 politeness suffixes that can be used in imp.p2. The problem is with сĂм, because it is added next to the stem before the person suffix (which exists only is p2.pl). For instance, for the verb кала in p2.pl we can have

калӑр ("normal" form without any politeness suffix)

But with сӑм:

кала.сӑм.ӑр

Other options are simple, since suffixes/clitics ччӗ and ха are added after the person suffix:

калӑрччӗ
каласӑмӑрччӗ
калӑр-ха
калӑрччӗ-ха
каласӑмӑрччӗ-ха

**Comment by jonorthwash** [comment]
If I understand right that the problem is that the morphemes are not in the order that you want the tags to be in, then you have a few options:
1. You can hard-code all the possible combinations (probably only a couple?)  e.g., `%<p2%>%<pl%>%<pol%>:%>с%{Ӑ%}м%>%{Ӑ%}р` (or whatever), etc..
2. You can use flag diacritics.
3. You can use twoc for non-suffixational morphology.
4. You can use Fran's hfst-concatenate trick.

If there are only a few possible combinations, 1 is probably the best.

**Comment by jonorthwash** [comment]
Btw, the way we deal with these in some other languages (in terms of tags) are like follows:
- kir: `кетчи/кет<v><iv><imp><p2><sg>+чы<emph>`
- kaz: `кетші/кет<v><imp><p2><sg>+шы<emph>`
- tur: `gitsene/git<v><iv><imp><emph><p2><sg>`, `gitsenize/git<v><iv><imp><emph><p2><pl>`
- tat: `китсәнә/кит<v><imp><p2><sg>+сана<emph>`, `китче/ит<v><iv><imp><p2><sg>+чы<emph>`
- crh: `ketçi/ket<v><iv><imp><p2><sg>+çı<emph>`

It looks like `tur` is the odd one out here.

**Comment by jonorthwash** [comment]
Note also that the Turkish forms at least superficially look like (but may well etymologically be) "conditional" forms plus -A (perhaps like a modal particle or interjection?).

Tatar seems to have reanalysed and borrowed the construction from Turkish (or late Ottoman at least?) as a single suffix.  The relevant conditional form in Tatar is `китсәң` (with ң not н).

**Comment by hectoralos** [comment]
@jonorthwash , yes, there are only two possibilities. I thought the order in the tags should follow the actual order in the surface form. If not necessarily, for sure it is easier as you suggest. I'll do it this way.
Just one more question about the politeness suffixes сĂм and ччӗ. Do I create the label <pol>, as in your example, or do I use <emph>, like чы kaj шы in Kyrgyz and Kazakh, which seem to be cognates of Chuvash ччӗ?

**Comment by jonorthwash** [comment]
> I thought the order in the tags should follow the actual order in the surface form. If not necessarily, for sure it is easier as you suggest. I'll do it this way.

Actually, I was assuming you wanted to put the tags in a different order from the order of the morphemes, and all my suggestions were ways to help you do that.  Perhaps I don't understand what the original question was.

> Do I create the label , as in your example, or do I use , like чы kaj шы in Kyrgyz and Kazakh, which seem to be cognates of Chuvash ччӗ?

I'm not sure I understand the question.  Are you asking whether you should use just a tag, like `<emph>` or whether you should treat it as a clitic with its own lemma and a tag?

**Comment by hectoralos** [comment]
1. No, I did not want to put the tags in a different order. But, if it is sometimes acceptable to put tags in an order different to the one in which they appear in the surface form, I am ready to do this here. So, if acceptable, I think we can analyse `кала.сӑм.ӑр` as `кала<v><vt><imp><p2><pl>+сӑм<mod>` instead of something else like `кала<v><vt><imp><pol><p2><pl>`. The first option is similar to what we get for the semantically similar `калӑрччӗ` (`кала<v><vt><imp><p2><pl>+ччӗ<mod>` ).

2. My question is about the tag for the suffix of politeness `ччӗ` (and maybe `сӑм` too). In [your first reply](https://github.com/apertium/apertium-chv/issues/17#issuecomment-529169367) you use `<polit>` and in [the second one](https://github.com/apertium/apertium-chv/issues/17#issuecomment-529169546), I see that `<emph>` is used in other languages. Currently, `<mod>` is used in chv.twol. I don't have any preference for any of them. I just tried to ask which of the three would be the best here. My guess is `<emph>`.

**Comment by jonorthwash** [comment]
> 1. No, I did not want to put the tags in a different order. But, if it is sometimes acceptable to put tags in an order different to the one in which they appear in the surface form, I am ready to do this here.

We try to avoid it, but there are instances where it makes sense.  The main argument for it would be if it made a construction parallel to another construction that's semantically (although not morphologically) parallel.

> So, if acceptable, I think we can analyse `кала.сӑм.ӑр` as `кала<v><vt><imp><p2><pl>+сӑм<mod>` instead of something else like `кала<v><vt><imp><pol><p2><pl>`. The first option is similar to what we get for the semantically similar `калӑрччӗ` (`кала<v><vt><imp><p2><pl>+ччӗ<mod>` ).

That seems like a reasonable argument to me.  Let's see what @ftyers thinks.

> 2. My question is about the tag for the suffix of politeness `ччӗ` (and maybe `сӑм` too). In [your first reply](https://github.com/apertium/apertium-chv/issues/17#issuecomment-529169367) you use `<polit>` and in [the second one](https://github.com/apertium/apertium-chv/issues/17#issuecomment-529169546), I see that `<emph>` is used in other languages. Currently, `<mod>` is used in chv.twol. I don't have any preference for any of them. I just tried to ask which of the three would be the best here. My guess is `<emph>`.

Aha.  Yeah, per https://github.com/apertium/apertium-turkic/issues/15 , I kind of think we should change all `<emph>`, `<qst>`, and `<mod_*>` to just `<mod>`, standing for something like "modal particle" or probably more accurately, "clitic conveying some sort of pragmatic information".  We don't need to encode the emphatic, question, politeness, etc. semantics of all of these in the ***part of speech tag*** name, since we're treating them as clitics anyway.  The only argument for treating them differently is that their distribution may differ from one another in a single language, but that's normal of other parts of speech too.  What do @ftyers and @IlnarSelimcan think?  (I'd be willing to allow `<qst>` and unite everything else too.)

Also, I should mention that `<pol>` was just for demonstration purposes.  I wasn't advocating for that way of encoding it at all—I just didn't know what you had in mind and needed something to stand in for it while I was discussing the order.

**Comment by ftyers** [comment]
I'm fine with the proposal for `сӑм`. 

I'm also fine with changing `<mod_*>` and `<emph>` to just `<mod>`. I kind of like `<qst>` but am not going to argue for it heavily.

**Comment by hectoralos** [comment]
In commit [602713f](https://github.com/apertium/apertium-chv/commit/602713fc886d5a936d6d58730bf452152a9928d4) I have:
* removed `<emph>`
* corrected `сӑм`
* corrected `<imp><p3><pl>`

Tests for politeness clitics can be found in [кала's yaml.file](https://github.com/apertium/apertium-chv/blob/master/tests/v_кала_del.yaml).

I have not changed anything about `<quest>`, since we wait form @IlnarSelimcan 's opinion.

I don't consider myself enough competent for arguing about these questions, but I'd like that you take into account that many of the putative users of this analyser are people who are used to a specific description of the language. The further we get away from this description, the fewer users will feel comfortable with the tool and, consequently, the less they will use it. On the question whether the `<quest>` tag should disappear, this doesn't worry me much. More worrying is that many suffixes are considered to be postpositions, when the Chuvash grammars say otherwise. So, in case of doubt, and if there are not strong reasons for changing the "traditional" or "local" description of the language, I'd try to stay as closer to it as possible.

**Comment by ftyers** [comment]
@hectoralos in principle it should not be a problem to have a version of the analyser where we replace the Apertium Turkic style tags with ones more familiar to Chuvash linguists. In the worst case a Python script with some `replace()` could do it.
